In [102]:
import os
os.getcwd()

'/content'

In [103]:
os.listdir()

['.config', 'drive', 'spam.csv', 'sample_data']

In [104]:
os.listdir()

['.config', 'drive', 'spam.csv', 'sample_data']

In [105]:
import pandas as pd
df = pd.read_csv('spam.csv', encoding='latin-1')
df.head()

v1                                                 v2 Unnamed: 2  \
0   ham  Go until jurong point, crazy.. Available only ...        NaN   
1   ham                      Ok lar... Joking wif u oni...        NaN   
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...        NaN   
3   ham  U dun say so early hor... U c already then say...        NaN   
4   ham  Nah I don't think he goes to usf, he lives aro...        NaN   

  Unnamed: 3 Unnamed: 4  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN

In [106]:
df = df[['v1', 'v2']]
df.columns = ['label', 'message']
df.head()

label                                            message
0   ham  Go until jurong point, crazy.. Available only ...
1   ham                      Ok lar... Joking wif u oni...
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...
3   ham  U dun say so early hor... U c already then say...
4   ham  Nah I don't think he goes to usf, he lives aro...

In [107]:
df['label'].value_counts()

label
ham     4825
spam     747
Name: count, dtype: int64

In [108]:
df['label'] = df['label'].map({'ham': 0, 'spam': 1})

In [109]:
import re

def clean_text(text):
    text = text.lower()
    text = re.sub(r'\W', ' ', text)
    return text

df['message'] = df['message'].apply(clean_text)

In [110]:
from sklearn.feature_extraction.text import CountVectorizer

cv = CountVectorizer()
X = cv.fit_transform(df['message'])
y = df['label']

In [111]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

In [112]:
from sklearn.naive_bayes import MultinomialNB

model = MultinomialNB()
model.fit(X_train, y_train)

MultinomialNB()

In [113]:
from sklearn.metrics import accuracy_score, classification_report

y_pred = model.predict(X_test)
print("Accuracy:", accuracy_score(y_test, y_pred))
print(classification_report(y_test, y_pred))

Accuracy: 0.97847533632287
              precision    recall  f1-score   support

           0       0.99      0.99      0.99       965
           1       0.91      0.93      0.92       150

    accuracy                           0.98      1115
   macro avg       0.95      0.96      0.95      1115
weighted avg       0.98      0.98      0.98      1115



In [114]:
msg = ["Congratulations! You won a free recharge"]
msg = cv.transform(msg)
model.predict(msg)

array([1])

In this project, we successfully built a Spam SMS Detection system using Machine Learning.
The SMS messages were preprocessed and converted into numerical features using
CountVectorizer. A Multinomial Naive Bayes classifier was trained and achieved high
accuracy (~98%), effectively classifying spam and non-spam messages.
